aws/resource_aws_dynamodb_table_test.go

```diff
@@ -14,6 +16,67 @@ import (
 	"github.com/hashicorp/terraform/terraform"
 )
 
+func init() {
+	resource.AddTestSweepers("aws_dynamodb_table", &resource.Sweeper{
+		Name: "aws_dynamodb_table",
+		F:    testSweepDynamoDbTables,
+	})
+}
+
+func testSweepDynamoDbTables(region string) error {
+	client, err := sharedClientForRegion(region)
+	if err != nil {
+		return fmt.Errorf("error getting client: %s", err)
+	}
+	conn := client.(*AWSClient).dynamodbconn
+
+	prefixes := []string{
+		"TerraformTest",
+		"tf-acc-test-",
+		"tf-autoscaled-table-",
+	}
+
+	err = conn.ListTablesPages(&dynamodb.ListTablesInput{}, func(out *dynamodb.ListTablesOutput, lastPage bool) bool {
+		for _, tableName := range out.TableNames {
+			hasPrefix := false
+			for _, prefix := range prefixes {
+				if strings.HasPrefix(*tableName, prefix) {
+					hasPrefix = true
+				}
+			}
+			if !hasPrefix {
+				continue
+			}
+			log.Printf("[INFO] Deleting DynamoDB Table: %s", *tableName)
+
+			input := &dynamodb.DeleteTableInput{
+				TableName: tableName,
+			}
+			err := resource.Retry(1*time.Minute, func() *resource.RetryError {
+				_, err := conn.DeleteTable(input)
+				if err != nil {
+					// Subscriber limit exceeded: Only 10 tables can be created, updated, or deleted simultaneously
+					if isAWSErr(err, dynamodb.ErrCodeLimitExceededException, "simultaneously") {
+						return resource.RetryableError(err)
+					}
+					return resource.NonRetryableError(err)
+				}
+				return nil
+			})
```

Comment:
This could be decoupled into a separate function and reused in both resource and the sweeper here. 🚿

Reply:
I'd actually like to submit this in a separate PR, since I need to re-add the retry handler that happens from removing a table from a global table then immediately deleting the table:

```
--- FAIL: TestAccAwsDynamoDbGlobalTable_multipleRegions (89.23s)
    testing.go:573: Error destroying resource! WARNING: Dangling resources
        may exist. The full state and error is shown below.
        
        Error: Error applying: 1 error(s) occurred:
        
        * aws_dynamodb_table.us-east-1 (destroy): 1 error(s) occurred:
        
        * aws_dynamodb_table.us-east-1: ResourceInUseException: Attempt to change a resource which is still in use: Table is being deleted: tf-acc-test-mryz6
            status code: 400, request id: M916D78UO89CO2GHJ2RQT8T0HJVV4KQNSO5AEMVJF66Q9ASUAAJG
```

Reply:
Or I can do it in this one. Up to you 😄

Reply:
Ah, ok, no probs. Separate PR is ok.